ragstack/colbertbase/colbertbase/astra_db.py

```diff
@@ -0,0 +1,177 @@
+from typing import List
+from cassandra.cluster import Cluster
+from cassandra.auth import PlainTextAuthProvider
+from cassandra import InvalidRequest
+from cassandra.concurrent import execute_concurrent_with_args
+
+from .token_embedding import PassageEmbeddings
+
+def required_cred(cred: str):
+    if cred is None or cred == "":
+        raise ValueError("Please provide credentials")
+
+class AstraDB:
+    def __init__(
```

Comment:
what's the advantage of cassio ? 
(and if using cassandra-driver do we still need secure bundle for astra vector compatibility?)

Reply:
we want a single auth interface. Since it's likely that you will use colbert with the langchain/llama CassandraStore, you will likely call `cassio.init`. If we can leverage it here, it'd be better